packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,501 @@
+   */
+  pause(): () => void;
+}
+```
+
+## <a name="Dependencies"></a>Synchronizing and Dependencies
+When scheduling jobs, it is often necessary to run jobs after certain other jobs are finished. 
+This is done through the `dependencies` options in the `schedule()` method.
+
+These jobs will also be passed to the job being scheduled, through its context. This can be
+useful if, for example, the output of those jobs are of a known type, or have known side channels.
+
+An example of this would be a compiler that needs to know the output directory of other compilers
+before it, in a tool chain.
+
+### Dependencies
+When scheduling jobs, the user can add a `dependencies` field to the scheduling options. The
+scheduler will wait for those dependencies to finish before running the job, and pass those jobs 
+in the context of the job.
+
+### Accessing Dependencies
+Jobs are called with a `JobHandlerContext` as a second argument, which contains a
+`dependencies: Job<JsonValue>[]` member which contains all dependencies that were used when 
+scheduling the job. Those aren't fully typed as they are determined by the user, and not the job
+itself. They also can contain jobs that are not finished, and the job should use the `state`
+member of the job itself before trying to access its content.
+
+### Scheduler Sub Jobs
+The `JobHandlerContext` also contains a `scheduler` member which can be used to schedule jobs
+using the same scheduler that was used for the job. This allows jobs to call other jobs
+and wait for them to end.
+
+## Available Schedulers
+The Core Angular DevKit library provides 2 implementations for the `Scheduler` interface:
+
+## SimpleJobRegistry
+Available in the jobs namespace. A registry that accept job registration, and can also schedule
+jobs.
+
+```typescript
+import { jobs } from '@angular-devkit/core';
+
+const add = jobs.createJobHandler<number[], number>(
+  input => input.reduce((total, curr) => total + curr, 0),
+);
+
+// Register the job in a SimpleJobRegistry. Different registries have different API.
+const registry = new jobs.SimpleJobRegistry();
+registry.register(add, {
+  name: 'add',
+  input: { type: 'array', items: { type: 'number' } },
+  output: { type: 'number' },
+});
+
+registry.schedule('add', [1, 2, 3, 4]);
+```
+
+### Job Compatibility
+The SimpleJobRegistry also allows a job to "implement" another job's interface. When doing so, 
+the argument, inputs, outputs and channels will also validate against the schemas of the job being
+implemented (in addition to the schemas of the job itself).
+
+This is done by adding an `implements` key to the options when registering a job. The value of
+this key should be the name of the job to also implement, and that job should be already registered.
+
+Unfortunately, this is done during registration and the job implementation has no idea if it's
+used in this way. This is a limitation of the current system that is being looked at.
+
+## NodeModuleJobScheduler
```

Comment:
I think this is now `NodeModuleJobRegistry`.

Reply:
Done.